### PR TITLE
Hw9_task1

### PR DESCRIPTION
**Pull request by E1-Bart0**

### CODE REVIEW

File: All_home_works/hw9/task1.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+from typing import Iterator, List, Union
+
+
+def merge_sorted_files(file_list: List[Union[Path, str]]) -> Iterator:
+    num_generators = [_get_nums_from_file(file) for file in file_list]
+    numbers = [
+        (index, next(number, float("inf")))
+        for index, number in enumerate(num_generators)
+    ]
+    while not all(1 if num[1] == float("inf") else 0 for num in numbers):
```
Comment (Cjkjvfnby) [inline All_home_works/hw9/task1.py]: `1 if num[1] == float("inf") else 0` could we make it simpler?

File: test/hw9/test_task1.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+
+def test__get_nums_from_file(filename):
+    expected = [1, 2, 3]
+    result = list(_get_nums_from_file(filename))
+    assert result == expected
+
+
+@patch("All_home_works.hw9.task1._get_nums_from_file", return_value=iter([1, 2, 3, 4]))
+def test__merge_sorted_files__check_one_file(_get_nums_from_file):  # noqa: PT019
```
Comment (Cjkjvfnby) [inline test/hw9/test_task1.py]: What is PT019?

File: All_home_works/hw9/task1.py (hunk 1 above)
Comment (Cjkjvfnby) [inline All_home_works/hw9/task1.py]: Let's optimize it a bit for cases when we have many files, and files could have different lengths.

**Review by Cjkjvfnby — CHANGES_REQUESTED** [review]

File: All_home_works/hw9/task1.py (hunk 1 above)
Comment (Cjkjvfnby) [reply All_home_works/hw9/task1.py]: Can you describe the result of optimization.   Let's try to talk on big O notation.   n -> number of files, k - the average number of numbers in the file.  

What was before, what you get after?

Comment (E1-Bart0) [reply All_home_works/hw9/task1.py]: I understood my mistake and remade it

Comment (Cjkjvfnby) [reply All_home_works/hw9/task1.py]: Let's optimize it a bit for cases when we have many files, and files could have different lengths.

**Review by Cjkjvfnby — CHANGES_REQUESTED** [review]

File: All_home_works/hw9/task1.py [hunk 3]
```diff
@@ -16,24 +16,27 @@
...
-
-
-def _get_min_number(numbers, num_generators, stop):
-    index, num = min(numbers, key=lambda x: x[1])
-    next_number = next(num_generators[index], float("inf"))
-    numbers[index] = (index, next_number)
-    if next_number == float("inf"):
-        stop += 1
-    return num, stop
+    numbers__generators = _get_numbers_and_generators(file_list)
```
Comment (Cjkjvfnby) [inline All_home_works/hw9/task1.py]: It's more common to use a single underscore in names.  `snake_case` not `snake__case`.

**Review by Cjkjvfnby — APPROVED** [review]

**Pull request merged**